src/tools/compiletest/src/runtest.rs

```diff
@@ -2470,7 +2474,46 @@ impl<'test> TestCx<'test> {
     }
 
     fn run_rustdoc_json_test(&self) {
-        //TODO
+        //TODO: Add bless option.
+
+        assert!(self.revision.is_none(), "revisions not relevant here");
+
+        let out_dir = self.output_base_dir();
+        let _ = fs::remove_dir_all(&out_dir);
+        create_dir_all(&out_dir).unwrap();
+
+        let proc_res = self.document(&out_dir, true);
```

Comment:
Do you ever call `document(false)` when this is a JSON test? If not, I think you can check `self.config.mode == RustdocJson` to avoid repeating this info.

Reply:
Thats much nicer, Done in a1e82fac146a4a28130febe04f88126da9eaa0a3